mypy/checker.py

```diff
@@ -4865,6 +4860,56 @@ def replay_lookup(new_parent_type: ProperType) -> Optional[Type]:
             expr = parent_expr
             expr_type = output[parent_expr] = make_simplified_union(new_parent_types)
 
+    def refine_optional_in(self,
+                           item_type: Type,
+                           collection_type: Type,
+                           ) -> Optional[Type]:
+        """
+        Check whether a condition `optional_item in collection_type` can narrow away Optional.
```

Comment:
Probably you meant:

> Check whether a condition "**item_type** in collection_type" [...]

(There is no `optional_item` parameter for this docstring to refer to)